### DISCUSSION TITLE
does commit code have to be based on the master branch?

### DISCUSSION THREAD

**Discussion by yx-keith** [trinodb/trino#17827] [Q&A] [answered]
I did some optimization in trino-hudi connector on tag 406, if I want to commit it to trino community, do I modify based on the master branch, thanks!

**Comment by hashhar** [comment, marked as answer]
yes, all code is merged to master so all PRs should ideally be based on master otherwise there will be conflicts and test failures.

For such developer facing communication we prefer to use [Slack](https://trino.io/slack.html) - the #dev channel.

**Comment by yx-keith** [comment]
> yes, all code is merged to master so all PRs should ideally be based on master otherwise there will be conflicts and test failures.
> 
> For such developer facing communication we prefer to use [Slack](https://trino.io/slack.html) - the #dev channel.

thank you
